test/dtypes/test_nf4.py

```diff
@@ -176,6 +176,25 @@ def test_to_copy(self):
             inpt_tensor_bfloat16 = inpt_tensor_nf4.to(torch.bfloat16)
             torch.testing.assert_allclose(inpt_tensor, inpt_tensor_bfloat16, atol=0.13, rtol=0.13)
 
+    def test_to_bfloat16(self):
+        inpt_tensor = torch.rand(128, dtype=torch.bfloat16)
+        inpt_tensor_nf4 = to_nf4(inpt_tensor, 32, 2)
+        assert type(inpt_tensor_nf4) != torch.Tensor
+        assert type(inpt_tensor_nf4.to(torch.bfloat16)) == torch.Tensor
+
+    def test_linear(self):
+        import torchao
+        a = torch.randn(32, 32, dtype=torch.bfloat16, device='cuda')
+        a_nf4 = torchao.dtypes.to_nf4(a, 16, 2)
+        inp = torch.randn(2, 32, 32, dtype=a.dtype, device=a.device)
+        out1 = torch.nn.functional.linear(inp, a)
+        out2 = torch.nn.functional.linear(inp, a_nf4)
+        out3 = torch.compile(torch.nn.functional.linear, mode='max-autotune')(inp, a_nf4)
+
+        # torch.testing.assert_allclose(out1, out2)
+        # torch.testing.assert_allclose(out1, out3)
```

Comment:
Do these tests pass?

Reply:
Not without custom atol/rtol. But this happens in a test higher up. I need to refactor these tests a bit more before landing.